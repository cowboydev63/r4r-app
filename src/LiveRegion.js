import React from 'react';
import { connect } from 'react-redux';

// Use this component to announe state changes to screen-readers
// by passing a message to state/announcements/actions/newMessage()
class LiveRegion extends React.PureComponent {
    render() {
        return (
            <div 
                className="r4r-hidden"
                role="status"
                aria-live="polite" 
                aria-atomic="true" 
                tabIndex="-1"
            >
                {this.props.message}
            </div>
        )
    }
}

const mapStateToProps = ({ announcements }) => ({
    message: announcements.liveMessage,
})

export default connect(mapStateToProps)(LiveRegion);