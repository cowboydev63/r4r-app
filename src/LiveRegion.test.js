import React from 'react';
import { shallow } from 'enzyme';
import { LiveRegion } from './LiveRegion';

describe('LiveRegion Component', () => {
  it('renders correctly with a message prop', () => {
    const wrapper = shallow(<LiveRegion message="Hello!"/>);
    expect(wrapper).toMatchSnapshot();
  })
})