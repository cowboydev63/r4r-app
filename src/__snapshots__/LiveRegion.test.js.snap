// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveRegion Component renders correctly with a message prop 1`] = `
<Theme
  aria-atomic="true"
  aria-live="polite"
  className="r4r-hidden"
  element="div"
  role="status"
  tabIndex="-1"
>
  Hello!
</Theme>
`;
